share/doc/wake/logging.md

```diff
@@ -0,0 +1,64 @@
+# Logging
+
+This article gives a high level overview of how wake logging works.
+Check out [print.wake](https://github.com/sifive/wake/blob/master/share/wake/lib/core/print.wake)
+for commented code.
+
+## How Logging works on Wake Execution commands
+
+Every `Job` will emit a stderr and a stdout.
+
+When writing your wake code, you can control where each goes with `setPlanStderr` and `setPlanStdout` before you `runJob`.
+We call this which `stream` each goes to (we often refer to a `stream` as a `logLevel`).
+
+Regardless of which `stream` you direct the `Job`'s `Stderr` and `Stdout`,
+`wake` will store all the text in its database,
+discarding the information about the `stream` it was destined for at the time it was stored.
+It will however know whether it came from the `Job`'s stderr/stdout.
+
+So what's the difference between the different streams?
+What you see on your console when you run wake.
+It controls *what* you see (depending on the command line flags) and what it looks like (what color it is).
+It also controls which *actual* stream it is going to coming out of `wake`
+(stderr or stdout, or other file descriptors 3-5).
+
+There are a number of built in streams in wake which all start with `log...`.
+You can also make your own stream using `mkLogLevel`.
+You can use this to customize the color you want it to be shown as,
+or (more commonly) to grab output from a `Job` for passing to another `Job`,
```

Comment:
My understanding is that LogLevels don't affect this at all.

Reply:
the "this" you are referring to is "grabbing output from a `Job` for passing to another `Job`'?

Reply:
Yes. Sorry, the GitHub diff view for leaving comments really obfuscates which lines of code will actually be shown even when I specifically select just a single line to comment on. I guess there's no way to indicate that you really are commenting on one line.